packages/yoastseo/src/parse/structure/Token.js

```diff
@@ -9,7 +11,8 @@ class Token {
 	 * @param {SourceCodeRange} sourceCodeRange The start and end positions of the token in the source code.
 	 */
 	constructor( text, sourceCodeRange = {} ) {
-		this.text = text;
+		// Normalize single quotes so that tokens can be matched with strings regardless of single quote type.
+		this.text = normalizeSingle( text );
```

Comment:
I initially thought that modifying the text at this level might not be a good idea, because if at some points downstream we need information on the specific type of the single quote, we wouldn't have information. 
But the above case might only relevant for search-based highlighting. Right now I cannot think of other use cases. So I guess your approach is safe, and actually neater compared to normalizing the single quote on the fly when matching texts :)